packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -60,24 +60,134 @@ class StorageReference {
         <String, String>{'path': _pathComponents.join("/")});
   }
 
+  /// Retrieves metadata associated with an object at this [StorageReference].
+  Future<StorageMetadata> getMetadata() async {
+    return new StorageMetadata._fromMap(await FirebaseStorage._channel
+        .invokeMethod("StorageReference#getMetadata", <String, String>{
+      'path': _pathComponents.join("/"),
+    }));
+  }
+
   String get path => _pathComponents.join('/');
 }
 
+/// Metadata for a [StorageReference]. Metadata stores default attributes such as
+/// size and content type.
+class StorageMetadata {
+  StorageMetadata({
+    this.cacheControl,
+    this.contentDisposition,
+    this.contentEncoding,
+    this.contentLanguage,
+    this.contentType,
+  }) : bucket = null,
+       generation = null,
+       metadataGeneration = null,
+       path = null,
+       name = null,
+       sizeBytes = null,
+       creationTimeMillis = null,
+       updatedTimeMillis = null,
+       md5Hash = null;
+
+  StorageMetadata._fromMap(Map<String, dynamic> map)
+      : bucket = map['bucket'],
+        generation = map['generation'],
+        metadataGeneration = map['metadataGeneration'],
+        path = map['path'],
+        name = map['name'],
+        sizeBytes=  map['sizeBytes'],
+        creationTimeMillis = map['creationTimeMillis'],
+        updatedTimeMillis = map['updatedTimeMillis'],
+        md5Hash = map['md5Hash'],
+        cacheControl = map['cacheControl'],
+        contentDisposition = map['contentDisposition'],
+        contentLanguage = map['contentLanguage'],
+        contentType = map['contentType'],
+        contentEncoding = map['contentEncoding'];
+
+  /// The owning Google Cloud Storage bucket for the [StorageReference].
+  final String bucket;
+
+  /// A version String indicating what version of the [StorageReference].
+  final String generation;
+
+  /// A version String indicating the version of this [StorageMetadata].
+  final String metadataGeneration;
+
+  /// The path of the [StorageReference] object.
+  final String path;
+
+  /// A simple name of the [StorageReference] object.
+  final String name;
+
+  /// The stored Size in bytes of the [StorageReference] object.
+  final int sizeBytes;
+
+  /// The time the [StorageReference] was created.
+  final num creationTimeMillis;
+
+  /// The time the [StorageReference] was last updated.
```

Comment:
... in milliseconds since the epoch.
Should be an `int`. Same for the other time stamp.